overlord/snapstate/handlers.go

```diff
@@ -1203,6 +1203,44 @@ func (m *SnapManager) undoUnlinkCurrentSnap(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+func (m *SnapManager) doCreateSnapSave(t *state.Task, _ *tomb.Tomb) (err error) {
+	st := t.State()
+	st.Lock()
+	snapsup, _, err := snapSetupAndState(t)
+	st.Unlock()
+	if err != nil {
+		return err
+	}
+
+	// verify that ubuntu-save has been mounted under the expected path and
+	// that it is indeed a mount-point.
+	if hasSave, err := osutil.IsMounted(dirs.SnapSaveDir); err != nil || !hasSave {
+		if err != nil {
+			return fmt.Errorf("cannot check if ubuntu-save is mounted: %v", err)
+		}
+		return nil
+	}
+
+	saveDir := snap.CommonSaveDir(snapsup.InstanceName())
+	return os.MkdirAll(saveDir, 0755)
```

Comment:
usually this would be a Backend operation or included in another backend operation